### ISSUE TITLE
Add support for offset and flags

### ISSUE THREAD

**Issue by jadejr** [vChavezB/ble_utils#4]
I was using your library, but then I wanted to do long reads and writes and realized that the offset param and flagswere not passed though. 

You need flags at least for `write_cb` to support prepared writes.

I did a quick hack like this 
```diff
diff --git a/include/ble_utils/ble_utils.hpp b/include/ble_utils/ble_utils.hpp
index 37b5669..335fc55 100644
--- a/include/ble_utils/ble_utils.hpp
+++ b/include/ble_utils/ble_utils.hpp
@@ -62,10 +62,11 @@ public:
      * @return Number of bytes read, or in case of an error
      *          BT_GATT_ERR() with a specific BT_ATT_ERR_* error code.
      */
-    virtual ssize_t read_cb(void *buf, uint16_t len)
+    virtual ssize_t read_cb(void *buf, uint16_t len, uint16_t offset)
     {
         ARG_UNUSED(buf);
         ARG_UNUSED(len);
+        ARG_UNUSED(offset);
         return 0;
     };
 
@@ -78,10 +79,12 @@ public:
     *  @return Number of bytes written, or in case of an error
     *          BT_GATT_ERR() with a specific BT_ATT_ERR_* error code.
     */
-    virtual ssize_t write_cb(const void *buf,uint16_t len)
+    virtual ssize_t write_cb(const void *buf,uint16_t len, uint16_t offset, uint8_t flags)
     {
         ARG_UNUSED(buf);
         ARG_UNUSED(len);
+        ARG_UNUSED(offset);
+        ARG_UNUSED(flags);
         return 0;
     }
 
diff --git a/src/ble_utils.cpp b/src/ble_utils.cpp
index cba26fa..4a6eb8d 100644
--- a/src/ble_utils.cpp
+++ b/src/ble_utils.cpp
@@ -60,7 +60,7 @@ ssize_t Characteristic::_read_cb(struct bt_conn *conn,
     ARG_UNUSED(conn);
     ARG_UNUSED(offset);
     auto instance = static_cast<Characteristic *>(attr->user_data);
-    return instance->read_cb(buf, len);
+    return instance->read_cb(buf, len, offset);
 }
 ssize_t Characteristic::_write_cb(struct bt_conn *conn,
                             const struct bt_gatt_attr *attr,
@@ -70,10 +70,8 @@ ssize_t Characteristic::_write_cb(struct bt_conn *conn,
                             uint8_t flags)
 {
     ARG_UNUSED(conn);
-    ARG_UNUSED(offset);
-    ARG_UNUSED(flags);
     auto instance = static_cast<Characteristic *>(attr->user_data);
-    return instance->write_cb(buf, len);
+    return instance->write_cb(buf, len, offset, flags);
 }
 
 Service::Service(const bt_uuid *uuid):
```
and it worked, but I wonder if some of the work could be done in the base characteristic class itself instead, so I didn't attempt to make this PR worthy yet.

**Comment by vChavezB** [comment]
I think that should be the correct approach. The internal callbacks  (`_write_cb`, `_read_cb`) are initialized in the characteristic gatt attribute [here ](https://github.com/vChavezB/ble_utils/blob/44f06a327da8e1bac8febef349f7d97398ab709b/src/ble_utils.cpp#L42-L43). So the parameters could then be passed to the user callback.

Feel free to open a PR.

Just be sure to update the sample so it still works with the new parameters and some documentation comments regarding the new parameters `offset:` https://github.com/zephyrproject-rtos/zephyr/blob/2765d23775146c58b91c5602664308f777fb0346/include/zephyr/bluetooth/gatt.h#L140
`flags:` https://github.com/zephyrproject-rtos/zephyr/blob/2765d23775146c58b91c5602664308f777fb0346/include/zephyr/bluetooth/gatt.h#L158

**Comment by vChavezB** [comment]
solved in #5
